src/test/java/seedu/address/testutil/PersonBuilder.java

```diff
@@ -46,8 +46,12 @@ public PersonBuilder(Person personToCopy) {
         name = personToCopy.getName();
         email = personToCopy.getEmail();
         module = personToCopy.getModule();
-        assignmentList = personToCopy.getAssignments().asUnmodifiableObservableList();
         tags = new HashSet<>(personToCopy.getTags());
+
+        assignmentList = new ArrayList<>();
+        for (Assignment assignments: personToCopy.getAssignments()) {
+            assignmentList.add(assignments);
+        }
     }
```

Comment:
if i modify the assignment, the assignment for both personToCopy's assignment list and newly created assignment list will change?

Reply:
Yes, I believe so. Since the assignment objects are the same for both list, a change in one assignment will affect the other. For mark assignment, what I believe you can do is to do a ```cloneAssignment``` and slightly change the implementation of ```AssignmentBuilder(Assignment)``` to instantiate a separate assignment object. For line 53, you can just change it to ```assignmentList.add(AssignmentBuilder(assignments).build())```.

On my part, I will change the naming on line 52 because i think it is slightly incorrect ```Assignment assignments``` to ```Assignment assignment```.